l1-contracts/deploy-scripts/InteropLibrary.sol

```diff
@@ -0,0 +1,301 @@
+// import {IInteropCenter} from "contracts/interop/InteropCenter.sol";
+import {InteropCenter} from "contracts/interop/InteropCenter.sol";
+import {InteropCallStarter} from "contracts/common/Messaging.sol";
+import {InteroperableAddress} from "contracts/vendor/draft-InteroperableAddress.sol";
+import {AmountMustBeGreaterThanZero, ZeroAddress} from "contracts/common/L1ContractErrors.sol";
+import {DataEncoding} from "contracts/common/libraries/DataEncoding.sol";
+import {IERC7786GatewaySource} from "contracts/interop/IERC7786GatewaySource.sol";
+
+// 0xa5cea466
+error ArgumentsLengthNotIdentical();
+
+library InteropLibrary {
+    /*//////////////////////////////////////////////////////////////
+                               BUILDERS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Build the “second bridge” calldata:
+    ///         bytes.concat(NEW_ENCODING_VERSION, abi.encode(l2TokenAssetId, abi.encode(amount, receiver, fee)))
+    function buildSecondBridgeCalldata(
+        bytes32 l2TokenAssetId,
+        uint256 amount,
+        address receiver,
+        address maybeTokenAddress
+    ) internal pure returns (bytes memory) {
+        bytes memory inner = DataEncoding.encodeBridgeBurnData(amount, receiver, maybeTokenAddress);
+        return DataEncoding.encodeAssetRouterBridgehubDepositData(l2TokenAssetId, inner);
+    }
+
+    /// @notice Create a single Interop call to the L2 asset router with the 7786 "indirectCall" attribute set.
+    function buildSecondBridgeCall(
+        bytes memory secondBridgeCalldata
+    ) internal pure returns (InteropCallStarter memory) {
+        bytes[] memory callAttributes = new bytes[](1);
+        callAttributes[0] = abi.encodeCall(IERC7786Attributes.indirectCall, (uint256(0)));
+        return
+            InteropCallStarter({
+                to: InteroperableAddress.formatEvmV1(L2_ASSET_ROUTER_ADDR),
+                data: secondBridgeCalldata,
+                callAttributes: callAttributes
+            });
+    }
+
+    /// @notice Build a single InteropCallStarter with provided attributes for sending a call.
+    function buildSendCall(
+        uint256 destinationChainId,
+        address target,
+        address executionAddress,
+        address unbundlerAddress,
+        bytes memory data
+    ) internal pure returns (InteropCallStarter memory) {
+        bytes[] memory callAttributes = buildCallAttributes(false, executionAddress, unbundlerAddress);
+
+        return
+            InteropCallStarter({
+                to: InteroperableAddress.formatEvmV1(destinationChainId, target),
+                data: data,
+                callAttributes: callAttributes
+            });
+    }
+
+    /// @notice Build a single InteropCallStarter with provided attributes for sending a bundle of calls.
+    function buildBundleCall(
+        address target,
+        address executionAddress,
```

Comment:
executionAddress and unbundledAddress is a bundle level attribute, it should not be here. You should not call buildCallAndBundleAttributes, but buildOnlyCallAttributes